src/axom/quest/SamplingShaper.hpp

```diff
@@ -536,11 +365,19 @@ class SamplingShaper : public Shaper
   {
     AXOM_ANNOTATE_SCOPE("finalizeShapeQuery");
 
-    delete m_inoutSampler2D;
-    m_inoutSampler2D = nullptr;
+    m_inoutSampler2D.reset();
```

Comment:
I see what you mean about having to keep track of various implementation pointers.